go/runtime/registry/registry.go

```diff
@@ -221,33 +255,67 @@ func (r *runtime) stop() {
 	r.history.Close()
 }
 
-func (r *runtime) watchUpdates(ctx context.Context, ch <-chan *registry.Runtime, sub pubsub.ClosableSubscription) {
+func (r *runtime) watchUpdates(
+	ctx context.Context,
+	ch <-chan *roothash.AnnotatedBlock,
```

Comment:
Instead of watching all blocks, this could also just watch epoch transitions as those are less frequent so there would be less wakeups.